base_mixin_restrict_field_access/models/restrict_field_access_mixin.py

```diff
@@ -72,6 +72,61 @@ def read(self, fields=None, load='_classic_read'):
                         self._fields[field].null(self.env))
         return result
 
+    def read_group(self, cr, uid, domain, fields, groupby, offset=0, limit=None, context=None, orderby=False, lazy=True):
+        """
+        Remove inaccessible fields from 'fields', 'groupby' and 'orderby'.
+
+        If this removes all 'fields', return no records.
+        If this removes all 'groupby', group by first remaining field.
+        If this removes 'orderby', don't specify order.
+        """
+        sanitised_fields = [f for f in fields if self._restrict_field_access_is_field_accessible(cr, uid, [], f)]
+        if 'restrict_field_access' in sanitised_fields:
+            sanitised_fields.remove('restrict_field_access')
+        if not sanitised_fields:
+            return []
+
+        sanitised_groupby = []
+        groupby = [groupby] if isinstance(groupby, basestring) else groupby
```

Comment:
it might be simpler to do this in https://github.com/OCA/OCB/blob/8.0/openerp/models.py#L1941
Here, there's no chance in hell to do this per record, so this should be its own action for sure, I don't have an opinion currently if we need `read_group_groupby`, `read_group_order` and `read_group` or if one suffices for all of them